apps/nextjs/src/components/RouteGuard.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from "react";
+
+import { useSession } from "@acme/auth/src/useSession";
+
+const RouteGuard = () => {
+  const session = useSession();
+
+  useEffect(() => {
+    if (!session.loading && !session.user) {
+      window.location.href = "/login";
```

Comment:
We shouldn't use window.location.href in nextjs, use Router.push or Router.replace